PhysicsTools/PatAlgos/python/slimming/MicroEventContent_cff.py

```diff
@@ -55,6 +58,8 @@
         'keep *_TriggerResults_*_*', # for MET filters (a catch all for the moment, but ideally it should be only the current process)
         'keep patPackedCandidates_lostTracks_*_*',
         'keep HcalNoiseSummary_hcalnoise__*',
+        'keep recoCSCHaloData_CSCHaloData_*_*',
```

Comment:
this one is not saved in AOD. 
So, an application running on AOD will not be able to get the objects.
Since the product is needed mainly on data, it may make sense to add it to the AOD event content, otherwise it's lost.